src/sentry/shared_integrations/client/base.py

```diff
@@ -78,6 +85,41 @@ def finalize_request(self, prepared_request: PreparedRequest) -> PreparedRequest
         """
         return prepared_request
 
+    def _get_redis_key(self):
+        """
+        Returns the redis key for the integration or empty str if cannot make key
+        """
+        if not hasattr(self, "integration_id"):
+            return ""
+        if not self.integration_id:
+            return ""
+        return f"sentry-integration-error:{self.integration_id}"
+
+    def is_error(self, e: Exception) -> bool:
+        if type(e) is ConnectionError:
+            return True
+        if type(e) is Timeout:
+            return True
+        if type(e) is HTTPError:
```

Comment:
@chloeho7 I think we need reverse logic here. An error is considered an error unless it's a 429 or 500. Maybe we rename the function `is_considered_error` because by definition every exception is an error.